lib/AST/NameLookup.cpp

```diff
@@ -1312,6 +1312,16 @@ MemberLookupTable::MemberLookupTable(ASTContext &ctx) {
 }
 
 void MemberLookupTable::addMember(Decl *member) {
+   // Peer through macro expansions.
+   if (auto *med = dyn_cast<MacroExpansionDecl>(member)) {
+     auto expanded = evaluateOrDefault(med->getASTContext().evaluator,
```

Comment:
This will eagerly expand all freestanding macros in type or extension scope, right? Can we make it more lazy or is that a huge change?

Reply:
FYI, I'm okay with landing this way. We'll need to integrate laziness with the mechanism that checks to make sure a macro doesn't produce a declaration that it didn't pre-declare via `names`.

Reply:
Good suggestion! In 8a3bff2 and e65fc44, I required declaration macros to specify introduced names and used the declared names to guide macro expansions lazily. PTAL.

In a follow-up patch I'll need to change the parser to support compound names to match the proposal.